Install fails for Perl 5.20

## Description

This module causes an abort for my Sympa Gitlab job: https://salsa.debian.org/sympa-team/sympa/-/jobs/2352397
All other Perl versions tested work fine.

Failure:

```
Building and testing Crypt-OpenSSL-X509-1.910 ... cp X509.pm blib/lib/Crypt/OpenSSL/X509.pm
Running Mkbootstrap for Crypt::OpenSSL::X509 ()
chmod 644 X509.bs
/root/.plenv/versions/5.20.3/bin/perl5.20.3 "-Iinc" /root/.plenv/versions/5.20.3/lib/perl5/site_perl/5.20.3/ExtUtils/xsubpp  -typemap /root/.plenv/versions/5.20.3/lib/perl5/5.20.3/ExtUtils/typemap -typemap typemap  X509.xs > X509.xsc && mv X509.xsc X509.c
Warning: Aliases 'subject_hash' and 'hash' have identical values in X509.xs, line 422
cc -c  -I/usr/local/opt/openssl/include -I/usr/local/include/openssl -I/usr/include/openssl -I/usr/local/include/ssl -I/usr/local/ssl/include -D_REENTRANT -D_GNU_SOURCE -fwrapv -fno-strict-aliasing -pipe -fstack-protector -I/usr/local/include -D_LARGEFILE_SOURCE -D_FILE_OFFSET_BITS=64 -O2 -g -Wall -Werror   -DVERSION=\"1.910\" -DXS_VERSION=\"1.910\" -fPIC "-I/root/.plenv/versions/5.20.3/lib/perl5/5.20.3/x86_64-linux-thread-multi/CORE"   X509.c
In file included from /root/.plenv/versions/5.20.3/lib/perl5/5.20.3/x86_64-linux-thread-multi/CORE/perl.h:5199,
                 from X509.xs:2:
/root/.plenv/versions/5.20.3/lib/perl5/5.20.3/x86_64-linux-thread-multi/CORE/inline.h:196:1: error: 'S_croak_memory_wrap' defined but not used [-Werror=unused-function]
 S_croak_memory_wrap(void)
 ^~~~~~~~~~~~~~~~~~~
cc1: all warnings being treated as errors
make: *** [Makefile:344: X509.o] Error 1
! Installing Crypt::OpenSSL::X509 failed. See /root/.cpanm/work/1641722497.40/build.log for details. Retry with --force to force install it.
```
Expiring 3 work directories.
FAIL
Cleaning up file based variables
00:01
ERROR: Job failed: exit code 1

### Expected behaviour

Installs without failure.

### Actual behaviour

Install fails, see above.

## Operating system and version

Docker container hatsoftwares/sympa-perl-5.20:latest

## Crypt::OpenSSL::X509 version

1.910

--- a/maint/Makefile_header.PL
+++ b/maint/Makefile_header.PL
@@ -22,7 +22,13 @@ if ($^O eq 'MSWin32') {
   }
 }
 
-my $cc_option_flags = '-O2 -g -Wall -Werror';
+my $cc_option_flags = '-O2 -g';
+
+if ($Config::Config{cc} =~ /gcc/i) {
+  $cc_option_flags .= $ENV{AUTHOR_TESTING} ? ' -Wall -Werror' : ' -Wall';
+} else {
+  $cc_option_flags .= '';
+}
 
 if ($Config{gccversion} =~ /llvm/i) {
   if ( $^O eq 'darwin' && $Config{gccversion} =~ /LLVM 12.0.5/) {
@@ -41,7 +47,7 @@ if ($Config{gccversion} =~ /llvm/i) {
 }
 
 if ($Config{myuname} =~ /sunos|solaris/i) {
-  # Any SunStudio flags?
+  $args{OPTIMIZE} = $cc_option_flags;
 } else {
   $args{OPTIMIZE} = $cc_option_flags;
 }

